src/v/redpanda/admin/util.cc

```diff
@@ -36,4 +57,24 @@ bool get_boolean_query_param(
            || str_param == "1";
 }
 
+bool path_decode(const std::string_view& in, ss::sstring& out) {
```

Comment:
Looks good, i'm wondering if we should considering patching this to seastar as well since it seems to be a legitimate bug in the library

Reply:
I'm happy to give that a go if you think that they would be open to accepting in the `path_decode` next to the existing `url_decode`. I would think that url decoding is not seastar's main concern, but they already expose `url_decode` so maybe it's worth a shot.

Reply:
@graphcareful Think @pgellert should do it to our fork first and then try to upstream it?  Or maybe attempt the upstream and they accept it we pull it into our fork?

Reply:
I'd lean towards making the change on our fork then opening a PR upstream. If it gets tied up in discussion w/ the seastar devs, we still get the convenience of consuming it like proper seastar code. My $0.02.